src/WPQueryDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Set return type of various WP_Query methods.
+ */
+
+declare(strict_types=1);
+
+namespace PHPStan\WordPress;
+
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\Type;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+
+class WPQueryDynamicMethodReturnTypeExtension implements \PHPStan\Type\DynamicMethodReturnTypeExtension
+{
+
+    public function getClass(): string
+    {
+        return 'WP_Query';
+    }
+
+    public function isMethodSupported(MethodReflection $methodReflection): bool
+    {
+        return in_array($methodReflection->getName(), [
+            'get_posts',
+            'query',
+        ], true);
+    }
+
+    /**
+     * @see https://developer.wordpress.org/reference/classes/wp_query/#return-fields-parameter
+     */
+    public function getTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): Type
+    {
+        // Called without arguments
+        if (count($methodCall->args) === 0) {
+            return new ArrayType(new IntegerType(), new ObjectType('WP_Post'));
+        }
```

Comment:
https://developer.wordpress.org/reference/classes/wp_query/get_posts/ has no parameters.
The `query` method has one.
What do you think?

Reply:
`$query = new WP_Query( $args ); $query->get_posts()` will cause the database query to run twice

Reply:
Ah, right, meant another method then. What I wanted to point out is that args can be passed to the constructor.

Reply:
Ah I see, yes. I need to find some time to come back to this PR.

Reply:
@johnbillion How about finishing this PR?